src/mlpack/methods/naive_bayes/naive_bayes_classifier.hpp

```diff
@@ -169,59 +182,51 @@ class NaiveBayesClassifier
    * @param probabilities This will be filled with class probabilities for each
    *      point. Each row represents a point.
    */
+  template<typename MatType>
   void Classify(const MatType& data,
                 arma::Row<size_t>& predictions,
-                arma::mat& probabilities) const;
+                ModelMatType& probabilities) const;
```

Comment:
Should we introduce a template parameter for the probabilities, I guess a user might ask why we introduced ``ProbabilitiesVecType`` for:

```
template<typename VecType, typename ProbabilitiesVecType>
void Classify(const VecType& point,
              size_t& prediction,
              arma::vec& probabilities) const;
              ProbabilitiesVecType& probabilities) const;
```

but not for the matrix version?